qtp_job_output_folder/summary.py

```diff
@@ -0,0 +1,91 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import exists, isdir, join
+from glob import glob
+from json import dumps
+
+
+def _folder_listing(folder):
```

Comment:
@antgonza is this doing what you want? Running this locally I see that not all subdirectories are flagged as "folder"s. For example here's a directory structure I made up:

```
In [18]: !tree ./                                                                                                                                                                                                                             
./
└── rofl
    └── plel
        ├── er
        └── lol
            └── ing

3 directories, 2 files

In [20]: ls -R .                                                                                                                                                                                                                              
rofl/

./rofl:
plel/

./rofl/plel:
er   lol/

./rofl/plel/lol:
ing
```

`rofl`, `plel` and `lol` are directories and `er` and `ing` are files. Here's the output I see:

```
In [17]: _folder_listing('./')                                                                                                                                                                                                                
Out[17]: 
[('folder', './rofl'),
 ('folder', './rofl/plel/lol'),
 ('file', './rofl/plel/er')]
```

I would expect to see `./rofl/plel` as another entry but that doesn't seem to be the case. Am I missing something?

Reply:
This is an interesting one, I think there are 2 things going on:
- * vs *.*: I think we are fine not displaying the hidden folders
- linux vs mac: if you see the test that failed the difference is that in mac the empty folders do not get listed but in linux they do

Reply:
Thanks. There are no hidden folders in the examples above. Are the empty folders not listed because git can't track empty folders? i.e. those are not committed to the repo so they didn't make it to the Travis run.

In any case. If the output looks good to you then it should be fine. Also, just wanted to check does `os.walk` not help in this case?

Reply:
It does look good to me but I'm still surprised about the difference between mac and linux on point 2. In fact, the tests only pass on linux ... which again, IMOO is fine. Thank you for checking!